src/goose.rs

```diff
@@ -1222,6 +1204,29 @@ impl GooseClient {
                     }
                     raw_request.set_status_code(Some(status_code));
                     raw_request.set_final_url(r.url().as_str());
+
+                    // Load test client was redirected.
+                    if raw_request.url != raw_request.final_url {
+                        // Extract the base_url of the request made.
+                        let request_base_url = match Url::parse(&raw_request.url) {
+                            Ok(base_url) => base_url.to_string(),
+                            Err(e) => {
+                                error!("goose_send failed to parse requested URL: {}", e);
+                                std::process::exit(1);
+                            }
+                        };
+                        // If a request for the base_url was redirected, change base_url.
+                        if request_base_url == self.base_url.read().await.to_string() {
+                            let redirected_base_url = match Url::parse(&raw_request.final_url) {
+                                Ok(base_url) => base_url,
```

Comment:
... here not .to_string()?

Reply:
we store the base_url as a Url, not as a String.